### PR TITLE
Escaping also considers record structs

### PR DESCRIPTION
**Pull request by SteveDunn**
fix: https://github.com/dotnet/roslyn/issues/74117

Previously, `record struct` was not considered as needing to be escaped, so types like `record struct @decimal` would be represented as `record struct decimal`

### CODE REVIEW

**Comment by SteveDunn** [comment]
I'm not sure if those failures are related to my change, but anyone more familiar with them please correct me if I'm wrong.

File: src/Compilers/CSharp/Test/Symbol/SymbolDisplay/SymbolDisplayTests.cs [hunk 1]
```diff
@@ -1093,6 +1093,39 @@ class @true {
                 SymbolDisplayPartKind.Punctuation);
         }
 
+        [WorkItem(74117, "https://github.com/dotnet/roslyn/issues/74117")]
+        [Fact]
```
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Symbol/SymbolDisplay/SymbolDisplayTests.cs:1097]:
```suggestion
        [Fact, WorkItem(74117, "https://github.com/dotnet/roslyn/issues/74117")]
```

**Review by CyrusNajmabadi — APPROVED** [review]

**Review by 333fred — APPROVED** [review]

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for a second review.

File: src/Compilers/CSharp/Test/Symbol/SymbolDisplay/SymbolDisplayTests.cs [hunk 2]
```diff
@@ -1093,6 +1093,38 @@ class @true {
                 SymbolDisplayPartKind.Punctuation);
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/74117")]
+        public void TestBug74117()
```
Comment (cston) [inline src/Compilers/CSharp/Test/Symbol/SymbolDisplay/SymbolDisplayTests.cs:1097]: Consider using a descriptive name, perhaps `TestRecordStructName`.

**Review by cston — APPROVED** [review]

File: src/Compilers/CSharp/Test/Symbol/SymbolDisplay/SymbolDisplayTests.cs (hunk 2 above)
Comment (SteveDunn) [inline src/Compilers/CSharp/Test/Symbol/SymbolDisplay/SymbolDisplayTests.cs:1097]:
```suggestion
        public void TestRecordStructName()
```

Comment (SteveDunn) [reply src/Compilers/CSharp/Test/Symbol/SymbolDisplay/SymbolDisplayTests.cs]: Makes sense. From the few places that I looked in Roslyn that had the `WorkItem` attribute, the ones I spotted went with the `TestBug123` convention, but looking further, I can see more that don't follow that convention.

**Review by cston — APPROVED** [review]

**Pull request merged**